wa/workloads/antutu/__init__.py

```diff
@@ -109,16 +145,98 @@ def extract_scores(self, context, regex_version):
                             result = float('NaN')
                         entry = regex.pattern.rsplit(None, 1)[0]
                         context.add_metric(entry, result, lower_is_better=False)
+        #Calculate group scores if 'CPU' in entry:
+                        if 'CPU' in entry:
+                            cpu.append(result)
+                            cpu_result = sum(cpu)
+                        if 'GPU' in entry:
+                            gpu.append(result)
+                            gpu_result = sum(gpu)
+                        if any([i in entry for i in ['Data', 'Document', 'Image', 'User', 'Video']]):
+                            ux.append(result)
+                            ux_result = sum(ux)
+                        if any([i in entry for i in ['RAM', 'ROM']]):
+                            mem.append(result)
+                            mem_result = sum(mem)
                         expected_results -= 1
         if expected_results > 0:
             msg = "The Antutu workload has failed. Expected {} scores, Detected {} scores."
             raise WorkloadError(msg.format(len(regex_version), expected_results))
 
+        context.add_metric('CPU Total Score', cpu_result, lower_is_better=False)
+        context.add_metric('GPU Total Score', gpu_result, lower_is_better=False)
+        context.add_metric('UX Total Score', ux_result, lower_is_better=False)
+        context.add_metric('MEM Total Score', mem_result, lower_is_better=False)
+
+        #Calculate overall scores
+        overall_result = float(cpu_result + gpu_result + ux_result + mem_result)
+        context.add_metric('Overall Score', overall_result, lower_is_better=False)
+
     def update_output(self, context):
         super(Antutu, self).update_output(context)
+        if self.version.startswith('10'):
+            self.extract_scores(context, self.regex_matches_v10)
         if self.version.startswith('9'):
             self.extract_scores(context, self.regex_matches_v9)
         if self.version.startswith('8'):
             self.extract_scores(context, self.regex_matches_v8)
         if self.version.startswith('7'):
             self.extract_scores(context, self.regex_matches_v7)
+
+
+class AntutuBDP(Workload):
+
+    name = "antutu_bdp"
+    description = '''
+    Workload for executing the BDP versions of the Antutu APK.
+
+    This will only work with specific APKS provided by Antutu but does
+    unlock command line automation and the capturing of a result file
+    as opposed to using UiAuto and Regex.
+
+    Known working version: 10.4.3-domesticAndroidFullBdp
+    '''
+    package_name = 'com.antutu.ABenchMark'
+    test_dir = '/sdcard/Android/data/com.antutu/ABenchMark/files/.antutu/'
+    result_file = '/sdcard/Documents/antutu/last_result.json'
```

Comment:
Do we strictly need these hardcoded paths and will they be the same on all targets?

Could we make use of the `external_storage` and `external_storage_app_dir` target properties or something similar to generate these paths?

Reply:
I've done this but not sure if it's the correct way to do it. To use these properties I can't do it at the class level so I am reusing the same variables in each function that requires it.

Reply:
Thanks, I'd probably suggest setting this once either  in `__init__` or early in `initalize` and reusing the attribute in the subsequent locations e.g. similar to what we have in lmbench.

[1] https://github.com/ARM-software/workload-automation/blob/2d14c82f92f3b71089550671773c5718b7e4514a/wa/workloads/lmbench/__init__.py#L100C14-L100C24